pkg/plugins/golang/v3/edit.go

```diff
@@ -19,56 +19,47 @@ package v3
 import (
 	"fmt"
 
+	"github.com/spf13/afero"
 	"github.com/spf13/pflag"
 
 	"sigs.k8s.io/kubebuilder/v3/pkg/config"
 	"sigs.k8s.io/kubebuilder/v3/pkg/plugin"
 	"sigs.k8s.io/kubebuilder/v3/pkg/plugins/golang/v3/scaffolds"
-	"sigs.k8s.io/kubebuilder/v3/pkg/plugins/internal/cmdutil"
 )
 
+var _ plugin.EditSubcommand = &editSubcommand{}
+
 type editSubcommand struct {
 	config config.Config
 
 	multigroup bool
 }
 
-var (
-	_ plugin.EditSubcommand = &editSubcommand{}
-	_ cmdutil.RunOptions    = &editSubcommand{}
-)
-
-func (p *editSubcommand) UpdateContext(ctx *plugin.Context) {
-	ctx.Description = `This command will edit the project configuration. You can have single or multi group project.`
-
-	ctx.Examples = fmt.Sprintf(`# Enable the multigroup layout
-        %s edit --multigroup
+func (p *editSubcommand) UpdateMetadata(cliMeta plugin.CLIMetadata, subcmdMeta *plugin.SubcommandMetadata) {
+	subcmdMeta.Description = `This command will edit the project configuration.
+Features supported:
+	- Toggle between single or multi group projects.
+`
+	subcmdMeta.Examples = fmt.Sprintf(`# Enable the multigroup layout
+        %[1]s edit --multigroup
 
         # Disable the multigroup layout
-        %s edit --multigroup=false
-	`, ctx.CommandName, ctx.CommandName)
+        %[1]s edit --multigroup=false
+	`, cliMeta.CommandName)
```

Comment:
It is also a breaking change.

Reply:
Why is this a breaking change?

Reply:
It affects the consumer. You are changing the exported methods in many places.